Source/Core/Core/HW/EXI/EXI_DeviceMemoryCard.cpp

```diff
@@ -166,7 +166,9 @@ void CEXIMemoryCard::SetupGciFolder(u16 sizeMb)
   u32 CurrentGameId = 0;
   if (game_id.length() >= 4 && game_id != "00000000" &&
       SConfig::GetInstance().GetTitleID() != Titles::SYSTEM_MENU)
-    CurrentGameId = BE32((u8*)game_id.c_str());
+  {
+    CurrentGameId = Common::swap32(reinterpret_cast<const u8*>(game_id.c_str()));
```

Comment:
How does this work? Swapping endianness of a pointer?

Reply:
The swap functions have [overloads](https://github.com/dolphin-emu/dolphin/blob/master/Source/Core/Common/Swap.h#L114) for operating on the data pointed to by  u8 pointers.

Reply:
Oh, that is a bit counter-intuitive - so this in fact extracts first 4 characters of a string and byteswaps them, right?

Reply:
Yes.

Reply:
Cool - that said, I would personally prefer to see those not overloaded but rather as a family of `swapPtrXX` functions. I feel like right now it is needlessly easy to confuse `swap32(T)` and `swap32(T*)`.